src/appkit.cs

```diff
@@ -23721,4 +24387,592 @@ interface NSMenuValidation
+		[Export ("outlineOverlayStyle", ArgumentSemantic.Strong)]
+		NSScrubberSelectionStyle OutlineOverlayStyle { get; }
+
+		[Static]
+		[Export ("roundedBackgroundStyle", ArgumentSemantic.Strong)]
+		NSScrubberSelectionStyle RoundedBackgroundStyle { get; }
+
+		[Export ("makeSelectionView")]
+		NSScrubberSelectionView MakeSelectionView ();
+	}
+
+	[Mac (10,12,1)]
+	[BaseType (typeof(NSView))]
+	interface NSScrubber
+	{
+		[NullAllowed, Export ("dataSource", ArgumentSemantic.Weak)]
+		INSScrubberDataSource DataSource { get; set; }
+
+		[NullAllowed, Export ("delegate", ArgumentSemantic.Weak)]
+		INSScrubberDelegate Delegate { get; set; }
+
+		[Export ("scrubberLayout", ArgumentSemantic.Strong)]
+		NSScrubberLayout ScrubberLayout { get; set; }
+
+		[Export ("numberOfItems")]
+		nint NumberOfItems { get; }
+
+		[Export ("highlightedIndex")]
+		nint HighlightedIndex { get; }
+
+		[Export ("selectedIndex")]
+		nint SelectedIndex { get; set; }
+
+		[Export ("mode", ArgumentSemantic.Assign)]
+		NSScrubberMode Mode { get; set; }
+
+		[Export ("itemAlignment", ArgumentSemantic.Assign)]
+		NSScrubberAlignment ItemAlignment { get; set; }
+
+		[Export ("continuous")]
+		bool Continuous { [Bind ("isContinuous")] get; set; }
+
+		[Export ("floatsSelectionViews")]
+		bool FloatsSelectionViews { get; set; }
+
+		[NullAllowed, Export ("selectionBackgroundStyle", ArgumentSemantic.Strong)]
+		NSScrubberSelectionStyle SelectionBackgroundStyle { get; set; }
+
+		[NullAllowed, Export ("selectionOverlayStyle", ArgumentSemantic.Strong)]
+		NSScrubberSelectionStyle SelectionOverlayStyle { get; set; }
+
+		[Export ("showsArrowButtons")]
+		bool ShowsArrowButtons { get; set; }
+
+		[Export ("showsAdditionalContentIndicators")]
+		bool ShowsAdditionalContentIndicators { get; set; }
+
+		[NullAllowed, Export ("backgroundColor", ArgumentSemantic.Copy)]
+		NSColor BackgroundColor { get; set; }
+
+		[NullAllowed, Export ("backgroundView", ArgumentSemantic.Strong)]
+		NSView BackgroundView { get; set; }
+
+		[Export ("initWithFrame:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (CGRect frameRect);
+
+		[Export ("reloadData")]
+		void ReloadData ();
+
+		[Export ("performSequentialBatchUpdates:")]
+		void PerformSequentialBatchUpdates (Action updateBlock);
+
+		[Export ("insertItemsAtIndexes:")]
+		void InsertItemsAtIndexes (NSIndexSet indexes);
+
+		[Export ("removeItemsAtIndexes:")]
+		void RemoveItemsAtIndexes (NSIndexSet indexes);
+
+		[Export ("reloadItemsAtIndexes:")]
+		void ReloadItemsAtIndexes (NSIndexSet indexes);
+
+		[Export ("moveItemAtIndex:toIndex:")]
+		void MoveItemAtIndex (nint oldIndex, nint newIndex);
+
+		[Export ("scrollItemAtIndex:toAlignment:")]
+		void ScrollItemAtIndex (nint index, NSScrubberAlignment alignment);
+
+		[Export ("itemViewForItemAtIndex:")]
+		NSScrubberItemView GetItemViewForItemAtIndex (nint index);
+
+		[Export ("registerClass:forItemIdentifier:")]
+		void RegisterClass ([NullAllowed] Class itemViewClass, string itemIdentifier);
```

Comment:
We should have a RegisterClass variant that accepts a Type instead can be done with the WrapAttribute